src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -125,6 +128,31 @@ public void runAppCommand(List<String> args) throws ProcessRunnerException {
     logCommand(command);
     processRunner.setEnvironment(environment);
     processRunner.run(command.toArray(new String[command.size()]));
+    processRunner.run(command.toArray(new String[command.size()]));
+  }
+
+  /**
+   * @return true iff the specified component is installed in the local environment
+   */
+  public boolean isComponentInstalled(String id) throws ProcessRunnerException {
+    List<String> command = new ArrayList<>();
+    command.add(getGCloudPath().toString());
+    command.add("components");
+    command.add("list");
+    command.add("--format=json");
+    command.add("--filter=id:" + id);
+    
+    if (stdOutput == null) {
```

Comment:
I would love to do that. I think it might require major refactoring of the API though, but let me give it a bit more thought and see if I can crack it.